spec/support/jazz.rb

```diff
@@ -362,11 +362,29 @@ class EnsembleInput < GraphQL::Schema::InputObject
     argument :name, String, required: true
   end
 
+  class AddInstrument < GraphQL::Schema::Mutation
+    description "Register a new musical instrument in the database"
+
+    argument :name, String, required: true
+    argument :family, Family, required: true
+
+    field :instrument, InstrumentType, null: false
+
+    def resolve(name:, family:)
+      instrument = Jazz::Models::Instrument.new(name, family)
+      Jazz::Models.data["Instrument"] << instrument
+      { instrument: instrument }
+    end
+  end
+
   class Mutation < BaseObject
     field :add_ensemble, Ensemble, null: false do
       argument :input, EnsembleInput, required: true
     end
 
+    # TODO don't require `field` somehow
```

Comment:
We have a `MutationObjectType` class which handles this.

So you'd have `class Mutation < MutationObject`

Reply:
Interesting, do you have an equivalent to `field :doStuff, field: DoStuff.field`  ?

Reply:
`MutationObjectType` defines a `mutation` method:

```ruby
mutation :doStuff, DoStuff
```

However I think this confuses people who don't realize that a mutation is just a field. So I do prefer keeping `field`.

Reply:
> confuses people 

I was worried about the same thing, glad you feel that way too. Something like 

```
field :do_stuff, DoStuff
```

could work fine, but it doesn't give the reader a _good_ clue how `DoStuff` fits in the picture. But maybe not a big deal.

Reply:
There's a middle ground of `field :doStuff, mutation_field: DoStuff`

I think the main thing is just not having to manually call `DoStuff.field`